src/main/kotlin/stringCalculator/StringParser.kt

```diff
@@ -0,0 +1,21 @@
+package stringCalculator
+
+class StringParser {
+    fun parse(inputString: String): List<String> {
+
+        val customSeparator = checkCustomSeparator(inputString)
+            ?: return inputString.split(":", ",")
+
+        return inputString.replace("//$customSeparator\n", "").split(":", ",", customSeparator)
+    }
+
+    private fun checkCustomSeparator(inputString: String): String? {
+        val pattern = Regex(CUSTOM_SEPARATOR_PATTERN)
+        val matchResult = pattern.find(inputString)
+        return matchResult?.groupValues?.get(1)
+    }
```

Comment:
1. check라는 함수 이름은 일반적으로 어떤 값을 확인한 후 Unit이나 예외를 던지는 함수를 나타내요.(검증 시 사용하는 `check()` 함수를 생각해보시면 좋을 것 같아요) 이 함수는 문자열에서 커스텀 구분자를 추출하는 책임을 가졌으니 그 의미를 잘 드러내는 함수 이름으로 바꿔보면 어떨까요?
2. `Regex(CUSTOM_SEPARATOR_PATTERN)` 또한 변하지 않는 값이니 상수로 관리해주시면 좋을 것 같습니다!
3. 코틀린의 get 함수가 오퍼레이터라면 다음과 같이 사용할 수 있어요.
```kotlin
return matchResult?.groupValues[1]
```

Reply:
3번째 답변은 제가 잘못 답변했네요..🥲
non-null 타입에서만 get 오퍼레이터로 사용할 수 있군요.
그렇다면 다음과 같이 nullable 한 값을 위에서 빠르게 제거해주는 방식도 좋을 것 같아요.
```suggestion
         val matchResult = pattern.find(inputString) ?: return null
         return matchResult.groupValues[1]
```